Apps/CocoaPods-FCM/src/View/DashboardView.swift

```diff
@@ -1,8 +1,102 @@
 import CioDataPipelines
+import CioMessagingInApp
 import SwiftUI
+import UIKit
 import UserNotifications
 
+// ObservableObject that tracks the height of a subview
+class HeightTracker: ObservableObject {
+    @Published var subviewHeight: CGFloat = 0
+}
+
+// Used for associating the height tracker with InAppMessageView
+private var heightTrackerKey: UInt8 = 0
+private var maxHeight: CGFloat = 0
+
+extension InAppMessageView {
+    var heightTracker: HeightTracker? {
+        get {
+            objc_getAssociatedObject(self, &heightTrackerKey) as? HeightTracker
+        }
+        set {
+            objc_setAssociatedObject(self, &heightTrackerKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    // Tracks and update the height
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+
+        print("Height of tracker \(String(describing: heightTracker?.subviewHeight))")
+
+        // Update maxHeight and notify heightTracker if the frame height changes
+        // The height fluctuates, increasing and then decreasing.
+        // To prevent this behavior, we constrain it to the maximum height achieved by the subview.
+        if frame.height >= maxHeight {
+            maxHeight = frame.height
+            heightTracker?.subviewHeight = frame.height
+        }
+    }
+}
+
+struct InAppMessageViewRepresentable: UIViewRepresentable {
+    var elementId: String
+    @Binding var containerWidth: CGFloat
```

Comment:
For the UIKit inline View, we assume that the width constraint is provided by the customer. I think the SwiftUI view should match that behavior where the width isn't part of constructing the SwiftUI View, but the width is instead set by the customer.

Reply:
This code is removed and outdated.